controllers/pattern_controller.go

```diff
@@ -147,6 +148,27 @@ func (r *PatternReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	if err != nil {
 		return r.actionPerformed(qualifiedInstance, "applying defaults", err)
 	}
+	r.logger.Info("Applied Defaults")
+
+	// MBP-184 Implementation
+	// Ensure that the openshift-gitops namespace is present
+	// This namespace gets created by the OpenShift GitOps operator
+	// when it is installed.
+	if haveNamespace(r.Client, applicationNamespace) {
+		// If there are Application status present reset the array
+		if len(qualifiedInstance.Status.Applications) > 0 {
+			qualifiedInstance.Status.Applications = nil
+		}
+
+		// Retrieve the Pattern Application Details and apply them to the CR
+		qualifiedInstance, err = r.applyPatternAppDetails(r.argoClient, qualifiedInstance)
+
+		if err != nil {
+			return r.actionPerformed(qualifiedInstance, "application pattern details", err)
+		}
```

Comment:
I'm not sure we should prevent progress due to an error constructing the status

Reply:
Possibly an argument for moving it further down?

Reply:
@beekhof Would it be sufficient to just log an INFO that says: "No application details to apply"?

Reply:
After removing the Update() call, yes.
Although we don't know there are no applications, only that we couldn't retrieve them, so `"Could not retrieve ArgoCD applications: $error_text"`